configure.ac

```diff
@@ -63,7 +63,13 @@ AC_ARG_ENABLE([werror],
 AC_ARG_ENABLE([debug],
     [AS_HELP_STRING([--enable-debug],
                     [Turn on debug output])],
-    [debug=$enableval], [debug=no])
+    [debug=yes], [debug=no])
+
+if test "x${debug}" = "xyes"; then
+  CFLAGS="-Wall -Werror -Wfatal-errors -O0 -g3"
```

Comment:
As far as I know, changing CFLAGS user variable here will break user's ability to change CFLAGS, and it is discouraged.
I think it is better to define new variable, say, DEBUGCFLAGS, and append it to AM_CFLAGS in various Makefile.am.  The existing example is WARNCFLAGS.